### DISCUSSION TITLE
Env Var Processor - Trimming Trailing or Leading Slashes

### DISCUSSION THREAD

**Discussion by cancan101** [symfony/symfony#58014] [Q&A]
Is there anyway in the env var processors to trim trailing or leading slashes (`/`)? I am looking to canonicalize a user supplied URL and want to be robust to the user supplying slashes.

For example if the is the setting:
```
https://%env(MY_HOSTNAME)%/%env(MY_PATH)%
```
I want to strip the `/` from the `MY_HOSTNAME` so that I don't send up with a double `/` in the URL.

The `TRIM` documented [here](https://symfony.com/doc/current/configuration/env_var_processors.html) looks to only strip whitespace.

**Comment by MatTheCat** [comment]
You can create your own processor: https://symfony.com/doc/current/configuration/env_var_processors.html#custom-environment-variable-processors

Not sure how a user-supplied URL could end up split into different environment variables though :sweat_smile:
